File: internal/domain/agent/agent.go
package agent

import (
	"time"

	"gorm.io/gorm"
)

// Agent Agent实体
type Agent struct {
	ID          string         `gorm:"primaryKey;type:varchar(36)" json:"id"`
	Name        string         `gorm:"type:varchar(100);not null" json:"name"`
	Description string         `gorm:"type:text" json:"description,omitempty"`
	Avatar      string         `gorm:"type:varchar(500)" json:"avatar,omitempty"`
	Skills      string         `gorm:"type:text" json:"skills"` // JSON array of skills
	Metadata    string         `gorm:"type:text" json:"metadata,omitempty"` // JSON metadata
	OwnerID     string         `gorm:"type:varchar(36);not null;index" json:"owner_id"`
	AccessToken string         `gorm:"type:varchar(100);not null" json:"-"`
	Status      string         `gorm:"type:varchar(20);default:'inactive'" json:"status"` // inactive, active, online, offline
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Agent) TableName() string {
	return "agents"
}

// AgentOrgMembership Agent-组织关系
type AgentOrgMembership struct {
	AgentID    string    `gorm:"primaryKey;type:varchar(36)" json:"agent_id"`
	OrgID      string    `gorm:"primaryKey;type:varchar(36)" json:"org_id"`
	Status     string    `gorm:"type:varchar(20);not null" json:"status"` // pending, approved, rejected
	ApprovedBy string    `gorm:"type:varchar(36)" json:"approved_by,omitempty"`
	JoinedAt   time.Time `json:"joined_at,omitempty"`
	CreatedAt  time.Time `json:"created_at"`
}

func (AgentOrgMembership) TableName() string {
	return "agent_org_memberships"
}

// JoinRequest 加入申请
type JoinRequest struct {
	ID        string         `gorm:"primaryKey;type:varchar(36)" json:"id"`
	AgentID   string         `gorm:"type:varchar(36);not null;index" json:"agent_id"`
	OrgID     string         `gorm:"type:varchar(36);not null;index" json:"org_id"`
	Status    string         `gorm:"type:varchar(20);default:'pending'" json:"status"` // pending, approved, rejected
	Reason    string         `gorm:"type:text" json:"reason,omitempty"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

func (JoinRequest) TableName() string {
	return "join_requests"
}

File: openAIMClient/Mac/openAIM/openAIM/Models/Agent.swift
//
//  Agent.swift
//  openAIM
//
//  Created by Claude on 2026/3/7.
//

import Foundation

/// AI Agent 模型
struct Agent: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let description: String?
    let skills: [String]
    let accessToken: String?
    let status: AgentStatus
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case skills
        case accessToken = "access_token"
        case status
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

/// Agent 状态
enum AgentStatus: String, Codable {
    case active = "active"
    case inactive = "inactive"
}

/// 创建 Agent 请求
struct CreateAgentRequest: Codable {
    let name: String
    let description: String?
    let skills: [String]
}

/// 更新 Agent 请求
struct UpdateAgentRequest: Codable {
    let name: String?
    let description: String?
    let skills: [String]?
    let status: AgentStatus?
}